src/html/angular-i18n-translations-extractor.ts

```diff
@@ -0,0 +1,108 @@
+    context: HtmlTranslationExtractionContext
+): void {
+    const i18nElementTranslation = element.attributes.find(attribute => attribute.name === I18N_ATTRIBUTE_NAME);
+
+    if (i18nElementTranslation) {
+        handleTranslationsOfElements(element, context, i18nElementTranslation);
+    }
+
+    const i18nAttributeTranslation = element.attributes.filter(attribute => I18N_ATTRIBUTE_REGEX.test(attribute.name));
+
+    handleTranslationsOfAttributes(element, context, i18nAttributeTranslation);
+}
+
+function handleTranslationsOfElements(
+    element: AngularElement,
+    context: HtmlTranslationExtractionContext,
+    attribute: Attribute
+): void {
+    const translationId = extractTranslationId(attribute, context);
+
+    if (element.texts.length > 0 && translationId) {
+        context.registerTranslation({
+            translationId: translationId,
+            defaultText: element.texts[0].text,
+            position: element.startPosition
+        });
+    } else if (element.texts.length === 0) {
+        context.emitError(`The element ${context.asHtml()} with attribute  ${attribute.name} is empty and is therefore missing the default translation.`, attribute.startPosition);
+    }
+}
+
+function handleTranslationsOfAttributes(
+    element: AngularElement,
+    context: HtmlTranslationExtractionContext,
+    i18nAttributes: Attribute[]
+): void {
+
+    for (const i18nAttribute of i18nAttributes) {
+        handleAttribute(element, context, i18nAttribute);
+    }
+}
+
+function handleAttribute(
+    element: AngularElement,
+    context: HtmlTranslationExtractionContext,
+    i18nAttribute: Attribute
+): void {
+    const translationId = extractTranslationId(i18nAttribute, context);
+    if (!translationId) {
+        return;
+    }
+    const attributeName = i18nAttribute.name.substr(`${I18N_ATTRIBUTE_NAME}-`.length);
+    const attribute = element.attributes.find(attribute => attribute.name === attributeName);
+
+    if (!attribute) {
+        context.emitError(`The element ${context.asHtml()} with ${i18nAttribute.name} is missing a corresponding ${attributeName} attribute.`, element.startPosition);
+        return;
+    }
+
+    const defaultText = attribute.value;
+
+    if (!defaultText) {
+        context.emitError(`The element ${context.asHtml()} with ${i18nAttribute.name} is missing a value for the corresponding ${attributeName} attribute.`, element.startPosition);
+        return;
+    }
+
+    context.registerTranslation({
+        translationId: translationId,
+        defaultText: defaultText,
+        position: i18nAttribute.startPosition
+    });
+}
+
+function extractTranslationId(attribute: Attribute, context: HtmlTranslationExtractionContext): string {
+    const index = attribute.value.indexOf(ID_INDICATOR);
+    if (index < 0) {
+        context.emitError(`The attribute ${attribute.name} on element ${context.asHtml()} attribute is missing the custom id indicator '${ID_INDICATOR}'.`, attribute.startPosition);
```

Comment:
Can this error message be improved to point out how the user can fix the issue? E.g. linking to an external reference?